[core] Failed build on Windows with source zip archive

I'm using Windows and I download PMD source as a zip
I'm working with JDK 11
I failed mvnw.cmd clean verify with the following irrelevant test:

`[ERROR] Failures:[ERROR]
YAHTMLRendererTest.testReportMultipleViolations:109 File
YAHTMLSampleClass1.html is different expected:<<html>[
    <head>
        <title>PMD - YAHTMLSampleClass1</title>
    </head>
    <body>
        <h2>Class View</h2>
        <h3 align="center">Class: YAHTMLSampleClass1</h3>
        <table border="" align="center" cellspacing="0" cellpadding="3">
        <tr><th>Method</th><th>Violation</th></tr>
        <tr><td></td><td><table
border="0"><tr><td><b>Rule:</b></td><td>Foo</td></tr><tr><td><b>Description:</b></td><td>blah</td></tr><tr><td><b>Line:</b></td><td>1
and 1</td></tr></table></td></tr>
        <tr><td></td><td><table
border="0"><tr><td><b>Rule:</b></td><td>Foo</td></tr><tr><td><b>Description:</b></td><td>blah</td></tr><tr><td><b>Line:</b></td><td>1
and 1</td></tr></table></td></tr>
        </table>
    </body></html>]> but was:<<html>[
    <head>
        <title>PMD - YAHTMLSampleClass1</title>
    </head>
    <body>
        <h2>Class View</h2>
        <h3 align="center">Class: YAHTMLSampleClass1</h3>
        <table border="" align="center" cellspacing="0" cellpadding="3">
        <tr><th>Method</th><th>Violation</th></tr>
        <tr><td></td><td><table
border="0"><tr><td><b>Rule:</b></td><td>Foo</td></tr><tr><td><b>Description:</b></td><td>blah</td></tr><tr><td><b>Line:</b></td><td>1
and 1</td></tr></table></td></tr>
        <tr><td></td><td><table
border="0"><tr><td><b>Rule:</b></td><td>Foo</td></tr><tr><td><b>Description:</b></td><td>blah</td></tr><tr><td><b>Line:</b></td><td>1
and 1</td></tr></table></td></tr>
        </table>
    </body>]/html>>`

**Affects PMD Version:** 
latest
**Rule:**
NA
**Description:**
I'm using Windows and I download source as a zip
**Code Sample demonstrating the issue:**
NA

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*
CLI

--- a/docs/pages/release_notes.md
+++ b/docs/pages/release_notes.md
@@ -56,6 +56,7 @@ This is a {{ site.pmd.release_type }} release.
 
 *   all
     *   [#1196](https://github.com/pmd/pmd/issues/1196): \[core] CPD results not consistent between runs
+    *   [#1462](https://github.com/pmd/pmd/issues/1462): \[core] Failed build on Windows with source zip archive
 *   apex
     *   [#1542](https://github.com/pmd/pmd/pull/1542): \[apex] Include the documentation category
     *   [#1546](https://github.com/pmd/pmd/issues/1546): \[apex] PMD parsing exception for Apex classes using 'inherited sharing' keyword

--- a/pmd-core/src/test/java/net/sourceforge/pmd/renderers/YAHTMLRendererTest.java
+++ b/pmd-core/src/test/java/net/sourceforge/pmd/renderers/YAHTMLRendererTest.java
@@ -12,6 +12,7 @@ import java.io.IOException;
 import java.io.InputStream;
 import java.nio.charset.StandardCharsets;
 import java.util.Arrays;
+import java.util.regex.Pattern;
 
 import org.apache.commons.io.IOUtils;
 import org.junit.After;
@@ -104,11 +105,16 @@ public class YAHTMLRendererTest extends AbstractRendererTst {
             try (FileInputStream in = new FileInputStream(new File(outputDir, file));
                     InputStream expectedIn = YAHTMLRendererTest.class.getResourceAsStream("yahtml/" + file)) {
                 String data = IOUtils.toString(in, StandardCharsets.UTF_8);
-                String expected = IOUtils.toString(expectedIn, StandardCharsets.UTF_8);
+                String expected = normalizeLineSeparators(IOUtils.toString(expectedIn, StandardCharsets.UTF_8));
 
                 assertEquals("File " + file + " is different", expected, data);
             }
         }
+    }
+
+    private static String normalizeLineSeparators(String s) {
+        return s.replaceAll(Pattern.quote(IOUtils.LINE_SEPARATOR_WINDOWS), IOUtils.LINE_SEPARATOR_UNIX)
+                .replaceAll(Pattern.quote(IOUtils.LINE_SEPARATOR_UNIX), PMD.EOL);
     }
 
     @Override

--- a/pmd-doc/src/test/java/net/sourceforge/pmd/docs/MockedFileWriter.java
+++ b/pmd-doc/src/test/java/net/sourceforge/pmd/docs/MockedFileWriter.java
@@ -8,8 +8,12 @@ import java.io.IOException;
 import java.nio.file.Path;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.regex.Pattern;
 
+import org.apache.commons.io.IOUtils;
 import org.apache.commons.lang3.StringUtils;
+
+import net.sourceforge.pmd.PMD;
 
 public class MockedFileWriter implements FileWriter {
     
@@ -42,5 +46,10 @@ public class MockedFileWriter implements FileWriter {
 
     public void reset() {
         data.clear();
+    }
+
+    public static String normalizeLineSeparators(String s) {
+        return s.replaceAll(Pattern.quote(IOUtils.LINE_SEPARATOR_WINDOWS), IOUtils.LINE_SEPARATOR_UNIX)
+                .replaceAll(Pattern.quote(IOUtils.LINE_SEPARATOR_UNIX), PMD.EOL);
     }
 }

--- a/pmd-doc/src/test/java/net/sourceforge/pmd/docs/RuleDocGeneratorTest.java
+++ b/pmd-doc/src/test/java/net/sourceforge/pmd/docs/RuleDocGeneratorTest.java
@@ -67,6 +67,11 @@ public class RuleDocGeneratorTest {
         });
     }
 
+    private static String loadResource(String name) throws IOException {
+        return MockedFileWriter.normalizeLineSeparators(
+                IOUtils.toString(RuleDocGeneratorTest.class.getResourceAsStream(name), StandardCharsets.UTF_8));
+    }
+
     @Test
     public void testSingleRuleset() throws RuleSetNotFoundException, IOException {
         RuleDocGenerator generator = new RuleDocGenerator(writer, root);
@@ -82,17 +87,14 @@ public class RuleDocGeneratorTest {
         assertEquals(3, writer.getData().size());
         FileEntry languageIndex = writer.getData().get(0);
         assertTrue(FilenameUtils.normalize(languageIndex.getFilename(), true).endsWith("docs/pages/pmd/rules/java.md"));
-        assertEquals(IOUtils.toString(RuleDocGeneratorTest.class.getResourceAsStream("/expected/java.md"), StandardCharsets.UTF_8),
-                languageIndex.getContent());
+        assertEquals(loadResource("/expected/java.md"), languageIndex.getContent());
 
         FileEntry ruleSetIndex = writer.getData().get(1);
         assertTrue(FilenameUtils.normalize(ruleSetIndex.getFilename(), true).endsWith("docs/pages/pmd/rules/java/sample.md"));
-        assertEquals(IOUtils.toString(RuleDocGeneratorTest.class.getResourceAsStream("/expected/sample.md"), StandardCharsets.UTF_8),
-                ruleSetIndex.getContent());
+        assertEquals(loadResource("/expected/sample.md"), ruleSetIndex.getContent());
 
         FileEntry sidebar = writer.getData().get(2);
         assertTrue(FilenameUtils.normalize(sidebar.getFilename(), true).endsWith("docs/_data/sidebars/pmd_sidebar.yml"));
-        assertEquals(IOUtils.toString(RuleDocGeneratorTest.class.getResourceAsStream("/expected/pmd_sidebar.yml"), StandardCharsets.UTF_8),
-                sidebar.getContent());
+        assertEquals(loadResource("/expected/pmd_sidebar.yml"), sidebar.getContent());
     }
 }

--- a/pmd-doc/src/test/java/net/sourceforge/pmd/docs/SidebarGeneratorTest.java
+++ b/pmd-doc/src/test/java/net/sourceforge/pmd/docs/SidebarGeneratorTest.java
@@ -55,6 +55,8 @@ public class SidebarGeneratorTest {
         }
         String yaml = new Yaml(options).dump(result);
 
-        assertEquals(IOUtils.toString(SidebarGeneratorTest.class.getResourceAsStream("sidebar.yml"), StandardCharsets.UTF_8), yaml);
+        String expected = MockedFileWriter.normalizeLineSeparators(
+                IOUtils.toString(SidebarGeneratorTest.class.getResourceAsStream("sidebar.yml"), StandardCharsets.UTF_8));
+        assertEquals(expected, yaml);
     }
 }

